web_sdk/web_test_utils/lib/skia_client.dart

```diff
@@ -0,0 +1,446 @@
+    final File keys = workDirectory.childFile('keys.json');
+    final File failures = workDirectory.childFile('failures.json');
+
+    await keys.writeAsString(_getKeysJSON());
+    await failures.create();
+    final String commitHash = await _getCurrentCommit();
+
+    final List<String> imgtestInitCommand = <String>[
+      _goldctl,
+      'imgtest', 'init',
+      '--instance', _instance,
+      '--work-dir', workDirectory
+        .childDirectory('temp')
+        .path,
+      '--commit', commitHash,
+      '--keys-file', keys.path,
+      '--failure-file', failures.path,
+      '--passfail',
+    ];
+
+    if (imgtestInitCommand.contains(null)) {
+      final StringBuffer buf = StringBuffer()
+        ..writeln('A null argument was provided for Skia Gold imgtest init.')
+        ..writeln('Please confirm the settings of your golden file test.')
+        ..writeln('Arguments provided:');
+      imgtestInitCommand.forEach(buf.writeln);
+      throw Exception(buf.toString());
+    }
+
+    final io.ProcessResult result = await process.run(imgtestInitCommand);
+
+    if (result.exitCode != 0) {
+      final StringBuffer buf = StringBuffer()
+        ..writeln('Skia Gold imgtest init failed.')
+        ..writeln('An error occurred when initializing golden file test with ')
+        ..writeln('goldctl.')
+        ..writeln()
+        ..writeln('Debug information for Gold:')
+        ..writeln('stdout: ${result.stdout}')
+        ..writeln('stderr: ${result.stderr}');
+      throw Exception(buf.toString());
+    }
+    _isInitialized = true;
+  }
+
+  /// Executes the `imgtest add` command in the goldctl tool.
+  ///
+  /// The `imgtest` command collects and uploads test results to the Skia Gold
+  /// backend, the `add` argument uploads the current image test. A response is
+  /// returned from the invocation of this command that indicates a pass or fail
+  /// result.
+  ///
+  /// The [testName] and [goldenFile] parameters reference the current
+  /// comparison being evaluated by the [FlutterPostSubmitFileComparator].
+  Future<bool> imgtestAdd(String testName, File goldenFile) async {
+    final List<String> imgtestCommand = <String>[
+      _goldctl,
+      'imgtest', 'add',
+      '--work-dir', workDirectory
+        .childDirectory('temp')
+        .path,
+      '--test-name', cleanTestName(testName),
+      '--png-file', goldenFile.path,
+    ];
+
+    final io.ProcessResult result = await process.run(imgtestCommand);
+
+    if (result.exitCode != 0) {
+      // We do not want to throw for non-zero exit codes here, as an intentional
+      // change or new golden file test expect non-zero exit codes. Logging here
+      // is meant to inform when an unexpected result occurs.
+      print('goldctl imgtest add stdout: ${result.stdout}');
+      print('goldctl imgtest add stderr: ${result.stderr}');
+    }
+
+    return true;
+  }
+
+  /// Executes the `imgtest init` command in the goldctl tool for tryjobs.
```

Comment:
nit: for consistency `` `goldctl` ``

Reply:
went through all docs in this file and cleaned up references to the framework and its classes.